### PR TITLE
[Feature][Api] Refactor org.apache.dolphinscheduler.api.controller.DataAnalysisController

### LINKED ISSUE #11754: [Feature][Api] Refactor org.apache.dolphinscheduler.api.controller.DataAnalysisController
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

https://github.com/apache/dolphinscheduler/issues/10257 Suggest refactor the backend api
Refactor org.apache.dolphinscheduler.api.controller.TaskInstanceController

Suggested optimization

- Content-Type change to application/json
- Add more example to api doc
- Improve the accuracy of error reporting
- The input and output parameters of API should use entity

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zzihh**
## Purpose of the pull request

Fix https://github.com/apache/dolphinscheduler/issues/11754

### CODE REVIEW

**Comment by Zzihh** [comment]
@SbloodyS Just accidentally closed the last PR.rebuild this.

**Comment by DarkAssassinator** [comment]
parameters -> DTO is a good refactor, but why all service return still keep `Result`? I think that `Result` is a bad partice for service impl. If failed, service just need throw `ServiceException`, no need update the `Result`, such as https://github.com/apache/dolphinscheduler/pull/12736/commits/f7caa6a92ea3e784db4ab0cbf96a00811a985c50. WDYT. cc @SbloodyS @caishunfeng

**Comment by Zzihh** [comment]
@SbloodyS @caishunfeng hello , Please help review that.

**Comment by Zzih** [comment]
https://github.com/apache/dolphinscheduler/pull/13031 This PR has been completed and merged, then I will close my PR.

**Comment by Zzihh** [comment]
@caishunfeng @SbloodyS   Please help review that. please please please!!

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataAnalysisService.java [hunk 1]
```diff
@@ -41,7 +42,7 @@ public interface DataAnalysisService {
      * @param endDate     end date
      * @return task state count data
      */
-    Map<String, Object> countTaskStateByProject(User loginUser, long projectCode, String startDate, String endDate);
+    Result countTaskStateByProject(User loginUser, long projectCode, String startDate, String endDate);
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataAnalysisService.java:45]: `Result` should be encapsulated in `Collonter` layer, not `service` layer

**Review by CalvinKirs — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataAnalysisService.java (hunk 1 above)
Comment (Zzihh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataAnalysisService.java:45]: But in the interface v2(path:src/main/java/org/apache/dolphinscheduler/api/controller/v2), In most cases, Result be encapsulated in service layer

Comment (CalvinKirs) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataAnalysisService.java:45]: We better be able to revisit this

Comment (davidzollo) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataAnalysisService.java:45]: there are too many cases about `Result`, we can discuss where to put the `Result` later

**Review by davidzollo — APPROVED** [review]
LGTM

**Comment by zhongjiajie** [comment]
> Result should be encapsulated in Collonter layer, not service layer

I agree with Calvin, we should avoid returning simple object instead of results in service, especially since your patch is refactoring for our codebase, we should try to make it better

**Pull request closed without merging**